node/core/pvf/worker/src/common.rs

```diff
@@ -124,6 +123,35 @@ pub fn cpu_time_monitor_loop(
 	}
 }
 
+/// Attempt to convert an opaque panic payload to a string.
+///
+/// This is a best effort, and is not guaranteed to provide the most accurate value.
+pub fn stringify_panic_payload(payload: Box<dyn Any + Send + 'static>) -> String {
+	match payload.downcast::<&'static str>() {
+		Ok(msg) => msg.to_string(),
+		Err(payload) => match payload.downcast::<String>() {
+			Ok(msg) => *msg,
+			// At least we tried...
+			Err(_) => "unknown panic payload".to_string(),
+		},
+	}
+}
+
+/// Helper function to notify the thread waiting on this condvar. This follows the conventions in
+/// the std docs, including the use of a `pending` flag.
+pub fn cond_notify_all(cond: Arc<(Mutex<bool>, Condvar)>) {
+	let (lock, cvar) = &*cond;
+	let mut pending = lock.lock().unwrap();
+	*pending = false;
+	cvar.notify_all();
```

Comment:
nit: I guess this can be `notify_one`, I'm not sure if there's a performance reason to use one over the other?

Reply:
There's only single thread waiting for notification, you can do `notify_one`.
Doubt it will make any difference though.